website/src/blogs/2018-08-14-provisioning-google-cloud-with-k8s-using-its-in-house-tool-kops.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Provisioning Google Cloud with k8s using it’s in-house tool, KOPS
+author: Harshvardhan Karn
+author_info: Harshvardhan Karn works at MayaData Inc. He is a public speaker, has talked in few local meetups and at two major conferences. In his free time, he likes to play Guitar, Netflix.
+date: 14-08-2018
+tags: Docker, Kubernetes, Openebs, Kops
+excerpt: Setting up and using a cluster in GCP offers a few significant advantages over using GKE. For instance, using GCP gives the user the liberty to use their custom binaries or a pure Open Source Kubernetes.
+---
+
+I am very excited for this post, as I have been working on this content for a few weeks. I might sound naive throughout this, mainly because I am, to be honest. Setting up and using a cluster in _GCP_ offers a few significant advantages over using _GKE_. For instance, using GCP gives the user the liberty to use their _custom binaries_ or a _pure Open Source Kubernetes_. Also, _GKE_ does not support modification or access to the master node, whereas a manually setup-ed k8s cluster over _VMs_ does.
+
+To gain the full understanding of Kubernetes, some people like to get their hands dirty for a course of interval. Luckily, we have a few more tools to get a Kubernetes cluster up and running in VM instances of **Google Cloud**. I find that tools like _kubernetes-incubator/kubespray , crosscloudci/cross-cloud_ and kops, are not very straightforward to use, but _kops, kubespray_ is somewhat close. _Cross-cloud_ on the other hand, has poor documentation and is not very stable. _kops_ here stands for ‘_Kubernetes Operations._’ To be honest, I find this to be a good tool to deploy the Cluster over Google Cloud Platform (**GCP**) or Amazon Web Services (**AWS**). I certainly would not say it is the best, but this tool is documented to the extent that one can use it. The original idea of KOPS was to create user a production ready cluster in **AWS**. Allowing it to provision the GCP is luxury since GCP already provides with GKE and AWS does not.
+
+### **Requirements**
+
+These items are required to deploy the production-ready k8s cluster in GCP:
+
+- **KOPS**
+
+```
+wget -O kops
+https://github.com/kubernetes/kops/releases/download/$(curl -s https://api.github.com/repos/kubernetes/kops/releases/latest | grep tag_name | cut -d '"' -f 4)/kops-linux-amd64
+
+chmod +x ./kops
+
+sudo mv ./kops /usr/local/bin/
+```
```

Comment:
```suggestion
- **KOPS**

    ```
    wget -O kops
    https://github.com/kubernetes/kops/releases/download/$(curl -s https://api.github.com/repos/kubernetes/kops/releases/latest | grep tag_name | cut -d '"' -f 4)/kops-linux-amd64

    chmod +x ./kops

    sudo mv ./kops /usr/local/bin/
    ```
```

![image](https://user-images.githubusercontent.com/75615087/124241230-309ac400-db39-11eb-88fb-6e4cd5c36a80.png)
it will come like this
![image](https://user-images.githubusercontent.com/75615087/124241303-43ad9400-db39-11eb-9983-541fb757ed7e.png)